staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -415,6 +416,26 @@ func IsEnvVarName(value string) []string {
 	return errs
 }
 
+// IsEnvVarValue tests if a string is a valid environment variable value.
+// this is follow syscall.Setenv
+// syscall/env_unix.go
+// syscall/env_windows.go
+func IsEnvVarValue(value string) []string {
+	var errs []string
+
+	// on Plan 9, null is used as a separator, eg in $path.
+	if runtime.GOOS == "plan9" {
+		return errs
+	}
+	for i := 0; i < len(value); i++ {
+		if value[i] == 0 {
+			errs = append(errs, "a valid value must can't contain null")
+			break
+		}
+	}
```

Comment:
You can simplify this by using e.g. `if strings.IndexByte(value, 0) >= 0`

Reply:
Thanks updated